pkg/sh/net.go

```diff
@@ -28,10 +51,16 @@ func DownloadHTTP(ctx *task.Context, url string, toPath string) error {
 	if err != nil {
 		return fmt.Errorf("failed issuing GET request: %v", err)
 	}
-	if res.Body == nil {
-		return errors.New("no body from the GET request")
-	}
 	defer res.Body.Close()
+	if res.StatusCode != 200 {
+		buf := strings.Builder{}
+		_, err := io.Copy(&buf, res.Body)
+		if err == nil {
+			err = errors.New(buf.String())
```

Comment:
This may not always produce the prettiest error messages... could easily be a gigantic blob of HTML for all we know... thoughts? Easy to remove.